demos/github-demo/src/isograph-components/__isograph/User/RepositoryConnection/entrypoint.ts

```diff
@@ -0,0 +1,205 @@
+import type {IsographEntrypoint, NormalizationAst, RefetchQueryNormalizationArtifactWrapper} from '@isograph/react';
+import {User__RepositoryConnection__param} from './param_type';
+import {User__RepositoryConnection__output_type} from './output_type';
+import readerResolver from './resolver_reader';
+const nestedRefetchQueries: RefetchQueryNormalizationArtifactWrapper[] = [];
+
+const queryText = 'query RepositoryConnection ($first: Int, $after: String, $id: ID!) {\
```

Comment:
There's a bug that generates `const queryText = 'mutation RepositoryConnection ...`, but only in this place.
I've manually replaced `mutation` with `query` for testing

Reply:
[that's what I'm referring to here](https://github.com/isographlabs/isograph/blob/3f50636d6d1a968ef0a88bf3f52f2d3ea81b769a/docs-website/docs/backlog.md?plain=1#L47)

I think this is due to `generate_entrypoint_artifacts_with_client_field_traversal_result` getting the first fetchable type: `schema.fetchable_types.iter().next()`, which can be a mutation.

Try adding

```
    pub fn query_root_operation_name(&self) -> RootOperationName {
        *self
            .fetchable_types
            .iter()
            .find(|(_, root_operation_name)| root_operation_name.0 == "query")
            .expect("Expected query to be found")
            .1
    }
```

and see if that fixes your issue.

Anyway, this is a weird issue and we should think about what a good solution looks like, becuase ☝️ is not a good solution 😂

Reply:
I've fixed this